sdk/go/common/resource/asset.go

```diff
@@ -486,18 +491,24 @@ func NewAssetArchive(assets map[string]interface{}) (*Archive, error) {
 }
 
 func NewPathArchive(path string) (*Archive, error) {
+	if path == "" {
+		return nil, errors.New("path cannot be empty when constructing a path archive")
+	}
 	a := &Archive{Sig: ArchiveSig, Path: path}
 	err := a.EnsureHash()
 	return a, err
 }
 
 func NewURIArchive(uri string) (*Archive, error) {
+	if uri == "" {
+		return nil, errors.New("uri cannot be empty when constructing a URI archive")
+	}
 	a := &Archive{Sig: ArchiveSig, URI: uri}
 	err := a.EnsureHash()
 	return a, err
 }
 
-func (a *Archive) IsAssets() bool { return a.Assets != nil }
+func (a *Archive) IsAssets() bool { return !a.IsPath() && !a.IsURI() }
```

Comment:
This should just be a.Assets != nil. Whenever we construct an Archive object we'll ensure it's a non-nil map.